From d9600e2dd1b35a6e74494c8c98ba399d06685041 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Thu, 27 Feb 2014 10:35:59 +0100
Subject: [PATCH 06/55] Drop SEGV handler

The generated tracebacks are mostly useless without debuginfo (which is likely
not present if the crash is not anticipated) and prevent ABRT from doing a
better job.

Signed-off-by: Lubomir Rintel <lkundrak@v3.sk>
---
 grubby.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/grubby.c b/grubby.c
index 440c6277935..649597ed92b 100644
--- a/grubby.c
+++ b/grubby.c
@@ -4211,21 +4211,6 @@ int addNewKernel(struct grubConfig * config, struct singleEntry * template,
     return 0;
 }
 
-static void traceback(int signum)
-{
-    void *array[40];
-    size_t size;
-
-    signal(SIGSEGV, SIG_DFL);
-    memset(array, '\0', sizeof (array));
-    size = backtrace(array, 40);
-
-    fprintf(stderr, "grubby received SIGSEGV!  Backtrace (%ld):\n",
-            (unsigned long)size);
-    backtrace_symbols_fd(array, size, STDERR_FILENO);
-    exit(1);
-}
-
 int main(int argc, const char ** argv) {
     poptContext optCon;
     const char * grubConfig = NULL;
@@ -4368,8 +4353,6 @@ int main(int argc, const char ** argv) {
 
     useextlinuxmenu=0;
 
-    signal(SIGSEGV, traceback);
-
     int i = 0;
     for (int j = 1; j < argc; j++)
 	i += strlen(argv[j]) + 1;
-- 
2.17.1

